mp2-v1/tests/common/cases/query.rs

```diff
@@ -235,26 +253,61 @@ async fn prove_query(
         let proving_tree = UpdateTree::from_paths(all_paths, *primary_value as Epoch);
         let planner = QueryPlanner {
             ctx,
-            genesis: table.genesis_block,
+            table: &table,
             query: query.clone(),
             pis: &pis,
-            tree: &table.row,
-            columns: table.columns.clone(),
+            settings: &settings,
         };
         let info = RowInfo {
             satisfiying_rows: row_keys.clone(),
         };
         prove_query_on_tree(planner, info, proving_tree, *primary_value).await?;
     }
     // do the same for the single index tree now
+    let mut planner = QueryPlanner {
+        ctx,
+        table: &table,
+        query: query.clone(),
+        pis: &pis,
+        settings: &settings,
+    };
+    let initial_epoch = table.index.initial_epoch();
     let current_epoch = table.index.current_epoch();
-    let all_paths = stream::iter(touched_rows.keys())
-        .then(|primary| async {
+    let block_range = {
+        let block_range = (query.min_block..query.max_block)
+            .filter(|&primary| {
+                (primary as Epoch) >= initial_epoch && (primary as Epoch) <= current_epoch
+            })
+            .collect_vec();
+        info!("found {} blocks in range", block_range.len());
+        if block_range.len() == 0 {
+            // no valid blocks in the query range, so we need to choose a block to prove
+            // non-existence
+            let to_be_proven_node = if (query.max_block as Epoch) < initial_epoch {
+                initial_epoch + 1
+            } else if (query.min_block as Epoch) > current_epoch {
+                current_epoch
+            } else {
+                bail!(
+                    "Empty block range to be proven for query bounds {}, {}, but no node
+                    to be proven with non-existence circuit was found. Something is wrong",
+                    query.min_block,
+                    query.max_block
+                );
+            } as BlockPrimaryIndex;
+            prove_non_existence_index(&mut planner, to_be_proven_node).await?;
+            vec![to_be_proven_node]
+        } else {
+            block_range
+        }
+    };
```

Comment:
This only covers the case where we have zero results accross the whole query. 
But what if there is hole in the middle like there are results for block 10 and 12 but not 11.